packages/office-ui-fabric-react/src/components/Button/examples/Button.Toggle.Example.tsx

```diff
@@ -1,38 +1,22 @@
 import * as React from 'react';
 import { DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
 
-export interface IButtonToggleExampleState {
-  muted: boolean;
-}
+export const ButtonToggleExample: React.FunctionComponent<IButtonProps> = props => {
+  const { disabled, checked } = props;
+  const [muted, setMuted] = React.useState<boolean>(false);
 
-export class ButtonToggleExample extends React.Component<IButtonProps, IButtonToggleExampleState> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      muted: false
-    };
-  }
-
-  public render(): JSX.Element {
-    const { disabled, checked } = this.props;
-
-    // tslint:disable:jsx-no-lambda
-    return (
-      <div>
-        <div>
-          <DefaultButton
-            data-automation-id="test"
-            allowDisabledFocus={true}
-            disabled={disabled}
-            toggle={true}
-            checked={this.state.muted || checked}
-            text={this.state.muted ? 'Volume Muted' : 'Volume Unmuted'}
-            onClick={() => {
-              this.setState({ muted: !this.state.muted });
-            }}
-          />
-        </div>
-      </div>
-    );
-  }
-}
+  return (
+    <DefaultButton
+      allowDisabledFocus={true}
+      disabled={disabled}
+      toggle={true}
+      checked={muted || checked}
+      text={muted ? 'Volume muted' : 'Volume unmuted'}
+      iconProps={{ iconName: muted ? 'Volume0' : 'Volume3' }}
+      // tslint:disable-next-line:jsx-no-lambda
+      onClick={() => {
```

Comment:
Probably best for this to be a `useCallback` to promote best practices.

Reply:
I'm still not convinced that `useCallback` is actually all that necessary/helpful for the average scenario (see also discussion on https://github.com/OfficeDev/office-ui-fabric-react/pull/10143#discussion_r314177944). I think we should all talk about this in person tomorrow.

Reply:
@ecraig12345 you might be right about `useCallback`. https://kentcdodds.com/blog/usememo-and-usecallback

However, here's where I am still on the fence: passing in constantly mutating values.

Imagine that `DefaultButton` is defined using [React.memo](https://reactjs.org/blog/2018/10/23/react-v-16-6.html#reactmemo). Now passing in the constantly regening callbacks will never benefit from the React.memo scenario.

Maybe the right approach is: don't memo anything in a component; it assumes the caller knows what they're doing. Let the caller memo, and when they do, they also must ensure that they don't pass in constantly mutating values. It's not easy to "just know" that though.

Constantly mutating:

```tsx
<DefaultButton iconProps={{ stuff }} /> 
```

Reply:
I'd seen that article too, which is part of what influenced my thinking on the issue.

Not totally sure what you mean by this part (except the "constantly mutating values," I get that):

> Maybe the right approach is: don't memo anything in a component; it assumes the caller knows what they're doing. Let the caller memo, and when they do, they also must ensure that they don't pass in constantly mutating values. It's not easy to "just know" that though.